packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -331,11 +331,21 @@ export class DocumentDeltaConnection
 	/**
 	 * Submits a new signal to the server
 	 *
-	 * @param message - signal to submit
+	 * @param content - Content of the signal.
+	 * @param targetClientId - When specified, the signal is only sent to the provided client id.
 	 */
-	public submitSignal(message: IDocumentMessage): void {
+	public submitSignal(content: IDocumentMessage, targetClientId?: string): void {
 		this.checkNotDisposed();
-		this.emitMessages("submitSignal", [[message]]);
+
+		if (targetClientId) {
+			this.logger.sendErrorEvent({
+				eventName: "SubmitSignalTargetedClient",
+				message:
+					"Sending signals to specific client ids is not supported. The signal will be broadcast to all clients",
```

Comment:
This logging looks questionable to me. I feel like one of the two should happen:
1. There is a way for client code to learn if particular end-point supports targeted signals. In such case, if driver does not support targeted signals, then using it should throw an usage error.
2. There is no way for client to learn if such capability is present. In such case, driver should silently use broadcast, and not issue any error telemetry.

Reply:
We can detect it via `IConnected.supportedFeatures` so I went with number 1.